src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -625,6 +625,68 @@ private static string UnescapeDriveColon(string fileUri)
             return sb.ToString();
         }
 
+        /// <summary>
+        /// Converts a file system path into a DocumentUri required by Language Server Protocol.
+        /// </summary>
+        /// <remarks>
+        /// When sending a document path to a LSP client, the path must be provided as a
+        /// DocumentUri in order to features like the Problems window or peek definition
+        /// to be able to open the specified file.
+        /// </remarks>
+        /// <param name="path">
+        /// A file system path. Note: if the path is already a DocumentUri, it will be returned unmodified.
+        /// </param>
+        /// <returns>The file system path encoded as a DocumentUri.</returns>
+        internal static string ConvertPathToDocumentUri(string path)
+        {
+            const string fileUriPrefix = "file:///";
+
+            if (path.StartsWith("untitled:", StringComparison.Ordinal))
+            {
+                return path;
+            }
+
+            if (path.StartsWith(fileUriPrefix, StringComparison.Ordinal))
+            {
+                return path;
+            }
+
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // On a Linux filesystem, you can have multiple colons in a filename e.g. foo:bar:baz.txt
+                string absoluteUri = new Uri(path).AbsoluteUri;
+
+                // First colon is part of the protocol scheme, see if there are other colons in the path
+                int firstColonIndex = absoluteUri.IndexOf(':');
+                if (absoluteUri.IndexOf(':', firstColonIndex + 1) >= 0)
+                {
+                    absoluteUri = new StringBuilder()
+                        .Append(absoluteUri, firstColonIndex + 1, absoluteUri.Length - firstColonIndex - 1)
+                        .Replace(":", "%3A")
+                        .Insert(0, absoluteUri.ToCharArray(0, firstColonIndex + 1))
+                        .ToString();
+                }
+
+                return absoluteUri;
+            }
+
+            // VSCode file URIs on Windows need the drive letter lowercase, and the colon
+            // URI encoded. System.Uri won't do that, so we manually create the URI.
+            var newUri = new StringBuilder(System.Web.HttpUtility.UrlPathEncode(path));
+            int colonIndex = path.IndexOf(':');
+            if (colonIndex > 0)
+            {
+                int driveLetterIndex = colonIndex - 1;
+                char driveLetter = char.ToLowerInvariant(path[driveLetterIndex]);
+                newUri
```

Comment:
I think this can also use that overload I mentioned:

```csharp
newUri.Replace(path[driveLetterIndex], driveLetter, driveLetterIndex, 1);
```

and
```csharp
newUri.Replace(":", "%3A", colonIndex, 1);
```

Reply:
these could be combined into one `Replace` call too in theory.

Reply:
I realize that you don't store this to a variable. Does it replace in-place?